File: mobile/android/app/src/main/java/com/cdbl/leavemanager/data/api/DelegationService.kt
package com.cdbl.leavemanager.data.api

import com.cdbl.leavemanager.data.model.DelegationsResponse
import retrofit2.Response
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Query

interface DelegationService {
    @GET("approvals/delegate")
    suspend fun getDelegations(
        @Header("Authorization") token: String,
        @Query("type") type: String = "mine",
        @Query("includeInactive") includeInactive: Boolean = false
    ): Response<DelegationsResponse>

    @DELETE("approvals/delegate")
    suspend fun revokeDelegation(
        @Header("Authorization") token: String,
        @Query("id") id: Int
    ): Response<Map<String, Any>>
}

File: mobile/ios/CDBLLeaveManager/CDBLLeaveManager/Core/Networking/DelegationService.swift
import Foundation

actor DelegationService {
    static let shared = DelegationService()

    private let client = APIClient.shared

    private init() {}

    func getDelegations(type: String = "mine", includeInactive: Bool = false) async throws -> DelegationsResponse {
        let endpoint = "approvals/delegate?type=\(type)&includeInactive=\(includeInactive)"
        return try await client.request(endpoint)
    }

    func revokeDelegation(id: Int) async throws -> APIResponse<Bool> {
        return try await client.request(
            "approvals/delegate?id=\(id)",
            method: .delete
        )
    }
}
